includes/class-wc-gateway-swedbank-pay-vipps.php

```diff
@@ -225,171 +241,123 @@ public function thankyou_page( $order_id ) {
-			'payment' => [
-				'operation'      => 'Purchase',
-				'intent'         => 'Authorization',
-				'currency'       => $currency,
-				'prices'         => [
-					[
-						'type'      => 'Vipps',
-						'amount'    => round( $amount * 100 ),
-						'vatAmount' => round( $info['vat_amount'] * 100 )
-					]
-				],
-				'description'    => apply_filters(
-					'swedbank_pay_payment_description',
-					sprintf(
-						__( 'Order #%s', WC_Swedbank_Pay::TEXT_DOMAIN ), $order->get_order_number() ),
-					$order
-				),
-				'payerReference' => $customer_uuid,
-				'userAgent'      => $order->get_customer_user_agent(),
-				'language'       => $this->culture,
-				'urls'           => [
-					'completeUrl'       => html_entity_decode( $this->get_return_url( $order ) ),
-					'cancelUrl'         => $order->get_cancel_order_url_raw(),
-					'callbackUrl'       => WC()->api_request_url( __CLASS__ ),
-					'termsOfServiceUrl' => $this->terms_url
-				],
-				'payeeInfo'      => [
-					'payeeId'        => $this->payee_id,
-					'payeeReference' => str_replace( '-', '', $order_uuid ),
-					'orderReference' => $order->get_order_number()
-				],
-				'riskIndicator'  => $this->get_risk_indicator( $order ),
-				'prefillInfo'    => [
-					'msisdn' => apply_filters( 'swedbank_pay_vipps_phone_format', $phone, $order )
-				],
-				'metadata'   => [
-					'order_id' => $order_id
-				],
-			]
-		];
-
-		// Add subsite
-		if ( ! empty( $this->subsite ) ) {
-			$params['payment']['payeeInfo']['subsite'] = $this->subsite;
-		}
-
+		// Process payment
 		try {
-			$result = $this->request( 'POST', '/psp/vipps/payments', $params );
-		} catch ( \Exception $e ) {
-			$this->log( sprintf( '[ERROR] Process payment: %s', $e->getMessage() ) );
+			$result = $this->core->initiateVippsPayment(
+				$order_id,
+				apply_filters( 'swedbank_pay_vipps_phone_format', $order->get_billing_phone(), $order )
+			);
+
+			// Save payment ID
+			update_post_meta( $order_id, '_payex_payment_id', $result['payment']['id'] );
+
+			return array(
+				'result'   => 'success',
+				'redirect' => $result->getOperationByRel( 'redirect-authorization' ),
+			);
+		} catch ( Exception $e ) {
 			wc_add_notice( $e->getMessage(), 'error' );
 
 			return false;
 		}
+	}
 
-		// Save payment ID
-		update_post_meta( $order_id, '_payex_payment_id', $result['payment']['id'] );
+	/**
+	 * Process Refund
+	 *
+	 * If the gateway declares 'refunds' support, this will allow it to refund
+	 * a passed in amount.
+	 *
+	 * @param int $order_id
+	 * @param float $amount
+	 * @param string $reason
+	 *
+	 * @return  bool|wp_error True or false based on success, or a WP_Error object
+	 */
+	public function process_refund( $order_id, $amount = null, $reason = '' ) {
+		$order = wc_get_order( $order_id );
+		if ( ! $order ) {
+			return false;
+		}
 
-		// Get Redirect
-		$redirect = self::get_operation( $result['operations'], 'redirect-authorization' );
+		// Full Refund
+		if ( is_null( $amount ) ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function is_null() is discouraged; use strict comparison "=== null" instead.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)